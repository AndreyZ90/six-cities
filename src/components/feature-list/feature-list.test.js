import React from 'react';
import TestRenderer from 'react-test-renderer';

import FeatureList from '@/components/feature-list/feature-list';

describe(`FeatureList component snapshot`, () => {
  test(`Should correctly render FeatureList component (type: apartament)`, () => {
    const tree = TestRenderer.create(
        <FeatureList
          type={`apartment`}
          bedrooms={3}
          maxAdults={5}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Should correctly render FeatureList component (type: house)`, () => {
    const tree = TestRenderer.create(
        <FeatureList
          type={`house`}
          bedrooms={3}
          maxAdults={5}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Should correctly render FeatureList component (type: hotel)`, () => {
    const tree = TestRenderer.create(
        <FeatureList
          type={`hotel`}
          bedrooms={3}
          maxAdults={5}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Should correctly render FeatureList component (type: apartament)`, () => {
    const tree = TestRenderer.create(
        <FeatureList
          type={`room`}
          bedrooms={3}
          maxAdults={5}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
