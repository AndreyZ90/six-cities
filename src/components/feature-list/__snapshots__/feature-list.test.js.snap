// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureList component snapshot Should correctly render FeatureList component (type: apartament) 1`] = `
<ul
  className="property__features"
>
  <li
    className="property__feature property__feature--entire"
  >
    Apartment
  </li>
  <li
    className="property__feature property__feature--bedrooms"
  >
    3 Bedrooms
  </li>
  <li
    className="property__feature property__feature--adults"
  >
    Max 5 adults
  </li>
</ul>
`;

exports[`FeatureList component snapshot Should correctly render FeatureList component (type: apartament) 2`] = `
<ul
  className="property__features"
>
  <li
    className="property__feature property__feature--entire"
  >
    Private Room
  </li>
  <li
    className="property__feature property__feature--bedrooms"
  >
    3 Bedrooms
  </li>
  <li
    className="property__feature property__feature--adults"
  >
    Max 5 adults
  </li>
</ul>
`;

exports[`FeatureList component snapshot Should correctly render FeatureList component (type: hotel) 1`] = `
<ul
  className="property__features"
>
  <li
    className="property__feature property__feature--entire"
  >
    Hotel
  </li>
  <li
    className="property__feature property__feature--bedrooms"
  >
    3 Bedrooms
  </li>
  <li
    className="property__feature property__feature--adults"
  >
    Max 5 adults
  </li>
</ul>
`;

exports[`FeatureList component snapshot Should correctly render FeatureList component (type: house) 1`] = `
<ul
  className="property__features"
>
  <li
    className="property__feature property__feature--entire"
  >
    House
  </li>
  <li
    className="property__feature property__feature--bedrooms"
  >
    3 Bedrooms
  </li>
  <li
    className="property__feature property__feature--adults"
  >
    Max 5 adults
  </li>
</ul>
`;
